### ISSUE TITLE
Array references bug

### ISSUE THREAD

**Issue by s-panferov** [mobxjs/mobx-state-tree#340]
Hello, thanks for the amazing library. I'm trying to use it my project, and I think I found a bug.

**How to reproduce**

```typescript
export const B = types.model('Block', {
  id: types.identifier(types.string)
})

export const S = types
  .model('Store', {
    blocks: types.array(B),
    blockRefs: types.array(types.reference(B))
  })
  .actions(self => {
    return {
      order() {
        self.blockRefs.move(0, 1)
      }
    }
  })

const a = S.create({
  blocks: [{id: '1'}, {id: '2'}],
  blockRefs: ['1', '2']
})

a.order(0, 1)
```

**Expected result:**

Should properly reorder references

**Current result:**

`Error: Converting circular structure to JSON` while reporting an error about wrong identifier type.

**Versions:**

```
    "mobx": "^3.2.2",
    "mobx-state-tree": "^0.10.3",
```

**Comment by mweststrate** [comment]
Is this specifically related to the `move` operation? In other words, does
const [a, b] = self.blockRefs; self.blockRefs.replace([b,a]) work ?

Op di 29 aug. 2017 om 08:21 schreef Stanislav Panferov <
notifications@github.com>:

**Comment by mattiamanzati** [comment]
@mweststrate It seems that dehenancer is not called on move operations.
I tried to add this on top of the reconciliation process and works like a charm; do you prefer to release a mobx fix or keep the workaround in MST?

```
    // Prepare new values, try to reconcile
    newValues.forEach((newValue, index) => {
        // for some reason, instead of newValue we got a node, fallback to the storedValue
        if (newValue instanceof Node) newValue = newValue.storedValue
```

**Comment by mweststrate** [comment]
Sounds like something that should be fixed in Mobx, but using above fix
with a comment to this issue is fine for now

Op di 29 aug. 2017 om 09:41 schreef Mattia Manzati <notifications@github.com

**Comment by s-panferov** [comment]
Thanks for the quick response. Could you please publish the hotfix?

**Comment by mattiamanzati** [comment]
Published in 0.11.0
